### ISSUE TITLE
Expr in columns is True

### ISSUE THREAD

**Issue by mrocklin** [dask-contrib/dask-expr#298]
Not sure what's going on here.  Not a huge deal though:

```python
In [1]: from dask_expr.datasets import timeseries

In [2]: df = timeseries()

In [3]: df.head()
Out[3]:
                        name    id         x         y
timestamp
2000-01-01 00:00:00   Ingrid  1023 -0.745347 -0.032617
2000-01-01 00:00:01   Oliver   973 -0.280432  0.715767
2000-01-01 00:00:02    Sarah  1008  0.629102  0.581916
2000-01-01 00:00:03  Michael   996 -0.112556  0.247828
2000-01-01 00:00:04   Ingrid  1058 -0.439967  0.958142

In [4]: df.x in df.columns
Out[4]: False

In [5]: df.expr.x in df.expr.columns
Out[5]: True
```

**Comment by phofl** [comment]
This goes back to 

``bool(expr1 == expr2)`` evaluating to ``True``. Applying ``bool`` onto the equality comparison of 2 expressions will evaluate to True every time

**Comment by mrocklin** [comment]
Maybe we don't define `__eq__` and force expr users to use `Eq(..., ...)`.  This is not atypical in other similar projects.

**Comment by phofl** [comment]
You mean not defining it at all or

```
def __eq__(self, other):
    return self._name == other._name
```

Not defining it at all would disallow using expressions as keys in dicts or adding them to sets (which is what we do in multiple places, maybe we should stop doing this though). This has them same issue as bool() raising (which is what pandas and dask/dask does)

**Comment by phofl** [comment]
Fixed by #307
